torchvision/transforms/functional_tensor.py

```diff
@@ -887,15 +887,32 @@ def _cast_squeeze_out(img: Tensor, need_cast: bool, need_squeeze: bool, out_dtyp
     return img
 
 
-def _apply_grid_transform(img: Tensor, grid: Tensor, mode: str) -> Tensor:
+def _apply_grid_transform(img: Tensor, grid: Tensor, mode: str, fill: Optional[List[float]]) -> Tensor:
 
     img, need_cast, need_squeeze, out_dtype = _cast_squeeze_in(img, [grid.dtype, ])
 
     if img.shape[0] > 1:
         # Apply same grid to a batch of images
         grid = grid.expand(img.shape[0], grid.shape[1], grid.shape[2], grid.shape[3])
+
+    # Append a dummy mask for customized fill colors, should be faster than grid_sample() twice
+    if fill is not None:
+        dummy = torch.ones((img.shape[0], 1, img.shape[2], img.shape[3]), dtype=img.dtype, device=img.device)
+        img = torch.cat((img, dummy), dim=1)
+
     img = grid_sample(img, grid, mode=mode, padding_mode="zeros", align_corners=False)
 
+    # Fill with required color
+    if fill is not None:
+        mask = img[:, -1:, :, :]  # N * 1 * H * W
+        img = img[:, :-1, :, :]  # N * C * H * W
+        mask = mask.expand_as(img)
+        fill_img = torch.tensor(fill, dtype=img.dtype, device=img.device).view(1, len(fill), 1, 1).expand_as(img)
+        if mode == 'nearest':
+            img[mask < 1e-3] = fill_img[mask < 1e-3]  # Leave some room for error
```

Comment:
I have a question here. Appologies if we've already discussed that. Technically, we have the mask as img dtype and we recreate boolean tensor. And we have to compare to 1 or 0.5, right ? 
```python
img[mask < 0.5]
# or
img[mask < 1.0]
```
why the threshold is `1-e3` here ?

Reply:
In nearest interpolation, the mask should be all 0/1. So anything between (0-1) should be right? I use this to compensate for possible rounding errors from CUDA or something.

Reply:
In nearest the result will be already rounded and we only have to get back bool mask values. I think we can either do `img[mask < 0.5]` or `mask = mask.bool()`

Reply:
Yes, that seems better and faster.